bpdv3/Assets/Scripts/Floor/TileController.cs

```diff
@@ -5,53 +5,57 @@
 // Handles the scrolling of the floor.
 public class TileController : MonoBehaviour {
 
-    [Header("Tile movement speed")]
-    [Range(0f, 10f)]
-    public float speed;
-    const int s_BeatToStartScroll = 6;
-    const int s_PositionToResetTileValue = 0;
-    const float s_PositionToEndScrolling = -1.25f;
-    const int s_BeatToEndScroll = 12;
-
-    public GameObject[] tilesArray;
-    public GameObject player;
-    
-
-    Vector3 direction;
-    public float distance;
+    #region Private variables
+    private const float s_distance = 1.25f;
+    private const int s_beatToStartScroll = 6;
+    private const float s_durationOfScrolling = 3f;
+    private const int s_positionToResetTileValue = 0;
+    private const float s_positionToEndScrolling = -1.25f;
+    private const int s_beatToEndScroll = 12;
+    private Vector3 _direction;
+    private Vector3 _tileMovement;
+    #endregion
+
+    #region Public variables
+    public GameObject[] TilesGOArray;    
+    public bool IsPlatformMoving;
+    #endregion
 
     // Use this for initialization
     void Start()
     {
-        direction = Vector3.back;
-        distance = direction.magnitude;
+        _direction = Vector3.back;
     }
 
     // Update is called once per frame
     void Update()
     {
-        if (SceneController.Instance.scrollBeatCount >= 6)
+        if (SceneController.Instance.scrollBeatCount >= s_beatToStartScroll)
```

Comment:
All of code in here seems like they control the same thing u can create a method for this and that method in Update